faker/providers/misc/__init__.py

```diff
@@ -91,7 +91,7 @@ def binary(cls, length=(1 * 1024 * 1024)):
 
         Default blob size is 1 Mb.
         """
-        return os.urandom(length)
```

Comment:
It also fails [this test for binary](https://github.com/joke2k/faker/blob/026e0577696ff21732e0c3ac529bbce30d723f0b/tests/factory.py#L358-L366) because `os.urandom()` supports having a length of `0` (which we consider valid) while `getrandbits()` doesn't support getting `0` random bits.